azkaban-common/src/main/java/azkaban/AzkabanCommonModule.java

```diff
@@ -72,9 +69,9 @@ public AzkabanCommonModule(final Props props) {
   protected void configure() {
     install(new AzkabanCoreModule(this.props));
     bind(Storage.class).to(resolveStorageClassType());
+    bind(AzkabanDataSource.class).to(resolveDataSourceType());
```

Comment:
Why not implement a provider that returns an instance of AzkabanDataSource?

Reply:
Never mind. I understand now. You would like to use Guice to construct the DataSource instances instead of using "new" in the provider.

Reply:
Yes. Thanks to Guice, we don't need to actually new a class. How powerful Guice is!